src/coreclr/vm/jitinterface.cpp

```diff
@@ -12996,7 +12996,7 @@ PCODE UnsafeJitFunction(PrepareCodeConfig* config,
         //DbgPrintf("Jitted Entry at" FMT_ADDR "method %s::%s %s size %08x\n", DBG_ADDR(nativeEntry),
         //          pszDebugClassName, pszDebugMethodName, pszDebugMethodSignature, sizeOfCode);
 
-        ClrFlushInstructionCache(nativeEntry, sizeOfCode);
```

Comment:
Do we need to fix the implementation of `ClrFlushInstructionCache` instead?

Reply:
That would be too big hammer. Almost all usages of it don't require flushing of cache on intel.

Reply:
Btw, this is needed only because we reuse code heap memory for dynamic methods. It took me a while to figure it out.

Reply:
> Almost all usages of it don't require flushing of cache on intel.

So it soft flush can be only used in places where the code was never executed before. I think that we have more of those, for example `Precode::Reset` should use also use the hard flush.

Would it make sense to add `bool codeNeverExecutedBefore` argument to `ClrFlushInstructionCache` and switch all places to call `ClrFlushInstructionCache` with the right true/false argument? We can then audit it more easily. Also, it would allow us to centralize the policy and do things like only do the hard flushes when emulating if it has material overhead.